components/esp_dns/examples/esp_dns_basic/main/esp_dns_example.c

```diff
@@ -0,0 +1,343 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Unlicense OR CC0-1.0
+ */
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include <unistd.h>
+#include "esp_log.h"
+#include "esp_system.h"
+#include "nvs_flash.h"
+#include "esp_event.h"
+#include "esp_timer.h"
+#include "lwip/opt.h"
+#include "protocol_examples_common.h"
+#include "esp_dns.h"
+#if defined(CONFIG_MBEDTLS_CERTIFICATE_BUNDLE)
+#include "esp_crt_bundle.h"
+#endif
+
+#define ENABLE_NATIVE_UDP 1
```

Comment:
Can these values be defined in kconfig?

Reply:
I added these macros for my own convinience while testing. This example is suppose to demonstrate usage of all the transport types. So I have removed all these macros.